src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitter.java

```diff
@@ -126,85 +144,92 @@ public void flushBuffer() {
      */
     @Override
     public List<TrackerPayload> getBuffer() {
-        return eventStore.getAllEvents();
+        return this.eventStore.getAllEvents();
     }
 
     /**
-     * Customize the emitter buffer size to any valid integer greater than zero.
+     * Customize the emitter batch size to any valid integer greater than zero.
      *
-     * @param bufferSize number of events to collect before sending
+     * @param batchSize number of events to collect before sending
      */
     @Override
-    public void setBufferSize(final int bufferSize) {
-        Preconditions.checkArgument(bufferSize > 0, "bufferSize must be greater than 0");
-        this.bufferSize = bufferSize;
+    public void setBatchSize(final int batchSize) {
+        Preconditions.checkArgument(batchSize > 0, "batchSize must be greater than 0");
+        this.batchSize = batchSize;
     }
 
     /**
-     * Gets the Emitter Buffer Size
+     * Gets the Emitter Batch Size
      *
-     * @return the buffer size
+     * @return the batch size
      */
     @Override
-    public int getBufferSize() {
-        return this.bufferSize;
+    public int getBatchSize() {
+        return this.batchSize;
     }
 
-    /**
-     * Checks if bufferSize is reached
-     *
-     * @return the new Runnable object
-     */
-    private Runnable getCheckForEventsToSendRunnable() {
-        return () -> {
-            while (!isClosing) {
-                if (eventStore.getSize() >= bufferSize) {
-                    drainEventsAndSend(getBufferSize());
-                }
-            }
-        };
-    }
-
-    private void drainEventsAndSend(int numberOfEvents) {
-        List<TrackerPayload> payloads = eventStore.removeEvents(numberOfEvents);
-        execute(getPostRequestRunnable(payloads));
+    public long getRetryDelay() {
```

Comment:
I don't think `retryDelay` needs a getter at all, let alone a public one. I couldn't think of a good way to test it being set correctly otherwise

Reply:
yeah, this is one of the cases where testing complicates things. It would be easier if the retry mechanism was [extracted](https://en.wikipedia.org/wiki/Strategy_pattern) from the Emitter in a separate component, so in that case you would test the separate component and the getRetryDelay there would make much more sense.

I think for now it's ok, we can improve it in future iterations.